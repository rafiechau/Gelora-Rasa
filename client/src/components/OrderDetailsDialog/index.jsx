import { Dialog, DialogTitle, DialogContent, DialogActions, Button, DialogContentText } from '@mui/material';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const OrderDetailsDialog = ({ open, onClose, order }) => {
  const isValidDate = order && Date.parse(order?.tanggalPembelian);
  const formattedDate = isValidDate
    ? new Intl.DateTimeFormat('id-ID', {
        year: 'numeric',
        month: 'long',
        day: 'numeric',
      }).format(new Date(order?.tanggalPembelian))
    : '';
  return (
    <Dialog open={open} onClose={onClose} maxWidth="sm" data-testid="order-details-dialog">
      <DialogTitle className={classes.dialogTitle} data-testid="order-details-title">
        <FormattedMessage id="app_title_order_details" />
      </DialogTitle>
      <DialogContent className={classes.dialogContent} data-testid="order-details-content">
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_order_name_events" />: {order?.eventName}
        </DialogContentText>
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_order_total_tickets" />: {order?.totalTickets}
        </DialogContentText>
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_order_total_pay" />: {order?.totalPay}
        </DialogContentText>
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_order_status_order" />: {order?.status}
        </DialogContentText>
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_order_date_order" />: {formattedDate}
        </DialogContentText>
      </DialogContent>

      <DialogActions>
        <Button onClick={onClose} data-testid="order-details-close-button">
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
};

OrderDetailsDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  order: PropTypes.object, // Anda mungkin perlu mendefinisikan shape untuk 'order'
};

export default OrderDetailsDialog;
