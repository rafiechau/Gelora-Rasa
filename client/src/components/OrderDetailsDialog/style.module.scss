$dark-grey: #333;
$border-radius: 5px;
.dialog-title {
    text-align: center;
    color: $color-white;
}
.dialog-content {
    padding-top: 20px;
}

.dialogContent {
    padding: 20px;
    border-radius: $border-radius;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    color: $dark-grey;
    line-height: 1.6;

    .dialogText {
        margin-bottom: 10px;
        font-size: 1rem;
    }
}