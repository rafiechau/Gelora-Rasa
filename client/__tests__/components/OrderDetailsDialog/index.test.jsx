import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import OrderDetailsDialog from '@components/OrderDetailsDialog';

// Contoh properti untuk OrderDetailsDialog
const mockOrderProps = {
  eventName: 'Test Event',
  totalTickets: 2,
  totalPay: '100000',
  status: 'Confirmed',
  tanggalPembelian: '2023-01-01T12:00:00',
};

const mockOnClose = jest.fn();

const messages = {
  app_title_order_details: 'Detail Order',
};

describe('OrderDetailsDialog', () => {
  beforeEach(() => {
    render(
      <IntlProvider locale="id-ID" messages={messages}>
        <OrderDetailsDialog open onClose={mockOnClose} order={mockOrderProps} data-testid="order-details-dialog" />
      </IntlProvider>
    );
  });

  it('renders correctly and matches snapshot', () => {
    expect(screen.getByTestId('order-details-dialog')).toBeInTheDocument();
    expect(screen.getByTestId('order-details-dialog')).toMatchSnapshot();
  });

  it('calls onClose when close button is clicked', () => {
    const closeButton = screen.getByText(/Close/i);
    fireEvent.click(closeButton);
    expect(mockOnClose).toHaveBeenCalled();
  });
});
