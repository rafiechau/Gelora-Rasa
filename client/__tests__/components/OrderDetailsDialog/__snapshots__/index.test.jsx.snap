// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailsDialog renders correctly and matches snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="order-details-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r0:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root dialogTitle css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        data-testid="order-details-title"
        id=":r0:"
      >
        Detail Order
      </h2>
      <div
        class="MuiDialogContent-root dialogContent css-ypiqx9-MuiDialogContent-root"
        data-testid="order-details-content"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_order_name_events
          : 
          Test Event
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_order_total_tickets
          : 
          2
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_order_total_pay
          : 
          100000
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_order_status_order
          : 
          Confirmed
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_order_date_order
          : 
          1 Januari 2023
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="order-details-close-button"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
